### PR TITLE
feat(invites): Get all pending Invites

### PR DESCRIPTION
**Pull request by Okeke12nancy**
Description:
Implement an endpoint to fetch all pending invites in the superadmin. A pending invite is defined as an invite that has not been accepted yet (isAccepted: false). The endpoint should return a list of all such invites.

Acceptance Criteria:

Endpoint:

Method: GET
URL: /api/v1/invites/pending
Functionality:

The endpoint should query the database for all invites where isAccepted is false.
Return a response with a status code of 200 OK upon successful retrieval of the data.
If no pending invites are found, return an empty array in the data field.

Tests:

Write unit tests to cover:
Successful retrieval of pending invites.
Scenario where no pending invites are found.
Error handling when an exception is thrown during database query.
Documentation:

Update the API documentation to include this new endpoint, detailing its functionality, expected inputs, and outputs.

### CODE REVIEW

**Comment by Okeke12nancy** [comment]
@buka4rill please review and merge

File: .vscode/settings.json [hunk 1]
```diff
@@ -10,9 +10,14 @@
...
-    "sideBar.border": "#333d2a",
-    "editorGroup.border": "#333d2a"
+    "commandCenter.border": "#e7e7e799",
+    "sash.hoverBorder": "#333d2a",
+    "statusBarItem.remoteBackground": "#1a1f15",
+    "statusBarItem.remoteForeground": "#e7e7e7",
+    "titleBar.activeBackground": "#1a1f15",
+    "titleBar.activeForeground": "#e7e7e7",
+    "titleBar.inactiveBackground": "#1a1f1599",
+    "titleBar.inactiveForeground": "#e7e7e799"
```
Comment (CFCIfe) [inline .vscode/settings.json:20]: Don't update this file. Untrack it

File: package.json [hunk 2]
```diff
@@ -33,7 +33,6 @@
     "postinstall": "npm install --platform=linux --arch=x64 sharp"
   },
   "dependencies": {
-    "@css-inline/css-inline-linux-x64-gnu": "^0.14.1",
```
Comment (CFCIfe) [inline package.json:36]: Return this line. 


    "@css-inline/css-inline-linux-x64-gnu": "^0.14.1",

File: src/database/data-source.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ const dataSource = new DataSource({
   username: process.env.DB_USERNAME,
   password: process.env.DB_PASSWORD,
   host: process.env.DB_HOST,
+  port: parseInt(process.env.DB_PORT || '5432'),
```
Comment (CFCIfe) [inline src/database/data-source.ts:14]: Don't touch this.
```suggestion
  port: process.env.DB_PORT,
```

File: src/modules/invite/invite.service.ts [hunk 4]
```diff
@@ -33,6 +33,51 @@ export class InviteService {
...
+  //     return {
+  //       status_code: HttpStatus.OK,
+  //       message: 'Successfully fetched pending invites',
+  //       data: invites,
+  //     };
+  //   } catch (error) {
+  //     throw new InternalServerErrorException(`Internal server error: ${error.message}`);
+  //   }
+  // }
+
```
Comment (CFCIfe) [inline src/modules/invite/invite.service.ts:48]: Delete orphaned comments

File: src/modules/invite/invite.service.ts
Comment (CFCIfe) [inline src/modules/invite/invite.service.ts:1]: Delete orphaned comments on these files

Comment (CFCIfe) [reply src/modules/invite/invite.service.ts:1]: Remove try...catch .There is a responseInterceptor. Just throw expected errors but leave Internal Server Error.

File: src/modules/invite/invite.service.ts [hunk 5]
```diff
@@ -33,6 +33,51 @@ export class InviteService {
...
+          isAccepted: invite.isAccepted,
+          isGeneric: invite.isGeneric,
+          organisation: invite.organisation,
+          email: invite.email,
+        };
+      });
+
+      // Return the response with the mapped data
+      const responseData = {
+        status_code: HttpStatus.OK,
```
Comment (CFCIfe) [inline src/modules/invite/invite.service.ts:70]: We send this by default. Remove it

File: src/modules/invite/invite.service.ts [hunk 6]
```diff
@@ -33,6 +33,51 @@ export class InviteService {
...
+        message: 'Successfully fetched pending invites',
+        data: pendingInvitesDto,
+      };
+
+      return responseData;
```
Comment (CFCIfe) [inline src/modules/invite/invite.service.ts:75]:
```suggestion
      return {
        message: 'Successfully fetched pending invites',
        data: pendingInvitesDto,
      };
```

File: src/modules/role/entities/role.entity.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { Permissions } from '../../permissions/entities/permissions.entity';
 
 @Entity({ name: 'roles' })
 export class Role extends AbstractBaseEntity {
-  @Column()
+  @Column({ default: 'Ezekiel' })
```
Comment (CFCIfe) [inline src/modules/role/entities/role.entity.ts:7]: No reason to do this.

File: src/modules/permissions/entities/permissions.entity.ts [hunk 8]
```diff
@@ -4,9 +4,8 @@ import { Role } from '../../../modules/role/entities/role.entity';
 
 @Entity()
 export class Permissions extends AbstractBaseEntity {
-  @Column()
+  @Column({ default: 'user' })
```
Comment (CFCIfe) [inline src/modules/permissions/entities/permissions.entity.ts:7]: No reason to do this.

**Review by CFCIfe — COMMENTED** [review]
Resolve all conversations after implementation is done.

File: src/database/data-source.ts
Comment (TheCodeGhinux) [inline src/database/data-source.ts:1]: You shouldn't be touching this

File: src/modules/invite/invite.controller.ts [hunk 9]
```diff
@@ -46,7 +46,11 @@ export class InviteController {
     const allInvites = await this.inviteService.findAllInvitations();
     return allInvites;
   }
-
+  @Get('invites/pending')
```
Comment (TheCodeGhinux) [inline src/modules/invite/invite.controller.ts:49]: You did not add API docs to this.

File: src/modules/invite/invite.service.ts [hunk 10]
```diff
@@ -33,6 +33,51 @@ export class InviteService {
...
+  //     const invites = await this.inviteRepository.find({ where: { isAccepted: false } });
+  //     return {
+  //       status_code: HttpStatus.OK,
+  //       message: 'Successfully fetched pending invites',
+  //       data: invites,
+  //     };
+  //   } catch (error) {
+  //     throw new InternalServerErrorException(`Internal server error: ${error.message}`);
+  //   }
+  // }
```
Comment (TheCodeGhinux) [inline src/modules/invite/invite.service.ts:47]: Remove commented codes

**Review by Okeke12nancy — COMMENTED** [review]
Made the changes

**Review by CFCIfe — APPROVED** [review]

**Pull request merged**